recipe/meta.yaml

```diff
@@ -27,17 +27,30 @@ requirements:
     - recommonmark
     - wheel >=0.31
 
-# For an implementation of unit tests which work except for Windows, see
-#    https://github.com/conda-forge/staged-recipes/commit/f3c5e9f
 test:
   imports:
     - pyscaffoldext
     - pyscaffoldext.markdown
   commands:
     - pip check
     - 'putup --help | grep markdown'
+    # PyScaffold uses Git, so add the necessary environment variables to configure it.
+    - >
+      GIT_COMMITTER_NAME="conda-forge tester"
+      GIT_COMMITTER_EMAIL="tester@testing.com"
+      GIT_AUTHOR_NAME="conda-forge tester"
+      GIT_AUTHOR_EMAIL="tester@testing.com"
+      pytest -m "not skip_on_conda_build" --cov-config setup.cfg
```

Comment:
Hi @maresb, I might have lost track of the commits by this time already, but I believe I added the  `skip_on_conda_build` just for the main repository in PyScaffold...

Maybe we could get around just by using `-k "not test_generated_extension"` or something like `-m "not system or not slow"`.

Reply:
Ah, ok, I got confused. Thanks for clarifying! Which do you prefer?

Reply:
I made a draft PR for `skip_on_conda_build` here: https://github.com/conda-forge/pyscaffold-feedstock/pull/35

Please let me know which command to use here and I'll change it.

BTW, may I add you as a maintainer for all the PyScaffold feedstocks?

Reply:
Thanks @maresb.
Yes feel free to add me to all PyScaffold feedstocks

Reply:
Just to log here (if people are still interested). Chris recently posted a comment on gitter that hints about a slightly (very very marginally) more elegant way for this issue with the environment variables:

> ```yaml
>   script: MY_VAR=xxx "{{ PYTHON }} -m pip install . -vv"
> ```
> or
> ```yaml
>   script:
>     - export MY_VAR=xxx
>     - "{{ PYTHON }} -m pip install . -vv"
> ```
> If you want it to build on windows as well you can do something like:
> ```yaml
>  script:
>    - export MY_VAR=xxx  # [unix]
>    - set MY_VAR=xxx  # [win] (syntax might be wrong)
>    - "{{ PYTHON }} -m pip install . -vv"
> ```

Reply:
I just noticed your comment. The problem with 
```
   - export MY_VAR=xxx  # [unix]
   - set MY_VAR=xxx  # [win] (syntax might be wrong)
```
is that the selectors `[platform]` are disallowed for `noarch` packages. Thus you're restricted very awkwardly to commands which are simultaneously valid for Windows and Linux. (Unless I am somehow mistaken.) Since Windows uses `set` and Linux uses `export` I think it's impossible to modify environment variables from a `noarch` package in a multi-platform safe way.

Reply:
Oooh, I had no idea.
Yeah, if no one replies to my issue asking about a OS-independent way of declaring env vars for the tests in the conda-build repo, I think we don't have an option.

Thank you very much for clarifying @maresb.